### PR TITLE
feat(launch): add an escape hatch (`disable_job_creation`) to disable automatic job creation

### LINKED ISSUE #4888: [CLI]:
### Describe the bug

I'm calling wandb.init in a python file in a nested directory structure. It's failing with the following error:

```ERROR   SenderThread:15661 [sender.py:send_artifact():1417] send_artifact: failed for artifact "job/job-git_github.com_BambooHR_bhr-ai-labs-ml-ops.git_src_com_bamboohr_ai_labs_ml_ops_annotation_projects_enps_sentiment_enps_auto_tag_to_wandb_loader.py": 400 Client Error: Bad Request for url: https://api.wandb.ai/graphql```

```python
run = wandb.init(
            project=project,
            config={
                "tag": classification_tag,
                "origin": "spreadsheet",
                "labeler": "HR",
            },
        )
```

```shell

```

### Additional Files

[run.zip](https://github.com/wandb/wandb/files/10562780/run.zip)

### Environment

WandB version: 0.13.9

OS: macos Monterey 12.6.2

Python version: 3.9.7

Versions of relevant libraries:

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by KyleGoyette**
Fixes WB-NNNN
Fixes #4888

Description
-----------
Provides a setting `disable_job_creation` and environment variable `WANDB_DISABLE_JOB_CREATION` to allow users to prevent job creation. In case it's causing issues

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/env.py [hunk 1]
```diff
@@ -409,3 +410,12 @@ def disable_git(env: Optional[Env] = None) -> bool:
...
         val = False if val.lower() == "false" else True
     return val
+
+
+def disable_job_creation(env: Optional[Env] = None) -> bool:
+    if env is None:
+        env = os.environ
+    val = env.get(DISABLE_JOB_CREATION, default="False")
+    if isinstance(val, str):
+        val = False if val.lower() == "false" else True
```
Comment (kptkin) [inline wandb/env.py:420]: technically, you don't need this part `_str_as_bool` in settings will normalize it.

File: wandb/env.py [hunk 2]
```diff
@@ -409,3 +410,12 @@ def disable_git(env: Optional[Env] = None) -> bool:
     if isinstance(val, str):
         val = False if val.lower() == "false" else True
     return val
+
+
+def disable_job_creation(env: Optional[Env] = None) -> bool:
+    if env is None:
+        env = os.environ
+    val = env.get(DISABLE_JOB_CREATION, default="False")
```
Comment (kptkin) [inline wandb/env.py:418]:
```suggestion
    env = env or os.environ
    value = env.get(DISABLE_JOB_CREATION, False)
```

File: wandb/env.py [hunk 3]
```diff
@@ -409,3 +410,12 @@ def disable_git(env: Optional[Env] = None) -> bool:
...
+    if isinstance(val, str):
+        val = False if val.lower() == "false" else True
+    return val
```
Comment (kptkin) [inline wandb/env.py:421]:
```suggestion
    return value
```

File: wandb/env.py [hunk 4]
```diff
@@ -409,3 +410,12 @@ def disable_git(env: Optional[Env] = None) -> bool:
     if isinstance(val, str):
         val = False if val.lower() == "false" else True
     return val
+
+
+def disable_job_creation(env: Optional[Env] = None) -> bool:
```
Comment (kptkin) [inline wandb/env.py:415]: you could completely remove this function and just do os.environ.get(...) in settings up to you

File: wandb/sdk/internal/sender.py [hunk 5]
```diff
@@ -1554,27 +1554,32 @@ def get_local_info(self) -> "LocalInfo":
         return local_info
 
     def _flush_job(self) -> None:
+        if (
+            self._job_builder.disable
```
Comment (kptkin) [inline wandb/sdk/internal/sender.py:1558]: what is the meaning of the disable attribute? does it make sense to have the `disable_job_creation` as part of it's definition?

**Review by kptkin — APPROVED** [review]
Awesome! Thanks for adding this setting!

File: wandb/sdk/internal/sender.py (hunk 5 above)
Comment (KyleGoyette) [reply wandb/sdk/internal/sender.py:1558]: Valid point

File: wandb/env.py (hunk 4 above)
Comment (KyleGoyette) [reply wandb/env.py]: Oh, wow, nice!

**Pull request merged**